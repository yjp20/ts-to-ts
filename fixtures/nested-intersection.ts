/** @model */
interface Address {
  street: string;
  city: string;
  country?: string;
}

/** @model */
interface Contact {
  email: string;
  phone?: string;
}

/** @model */
interface PersonDetails {
  name: string;
  age: number;
}

/** @model */
type CompleteProfile = PersonDetails & Contact & Address;
