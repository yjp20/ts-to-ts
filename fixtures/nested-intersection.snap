
<<<<<<<<<<< typescript <<<<<<<<<<<<
/** @model */
interface Address {
  street: string;
  city: string;
  country?: string;
}

/** @model */
interface Contact {
  email: string;
  phone?: string;
}

/** @model */
interface PersonDetails {
  name: string;
  age: number;
}

/** @model */
type CompleteProfile = PersonDetails & Contact & Address;
===================================
// nested-intersection.ts
model CompleteProfile {
  name: string,
  age: float64,
  email: string,
  phone?: string,
  street: string,
  city: string,
  country?: string
};

model Address {
  street: string,
  city: string,
  country?: string
};

model Contact {
  email: string,
  phone?: string
};

model PersonDetails {
  name: string,
  age: float64
};
>>>>>>>>>>> typespec >>>>>>>>>>>>>>